test/test-cases.setup.js

```diff
@@ -0,0 +1,62 @@
+import { basename, dirname, join } from 'path';
+import { readFile } from 'fs';
+import webpack from 'webpack';
+import I18nPlugin from '../src';
+
+export function processFile(entry, ...pluginOpts) {
+  const resolvedEntry = join(__dirname, 'test-cases', entry);
+  const resolvedOutput = join(dirname(resolvedEntry), `${basename(resolvedEntry, '.code.js')}.tmp.js`);
+
+  const compiler = webpack({
+    entry: resolvedEntry,
+    output: {
+      filename: basename(resolvedOutput),
+      path: dirname(resolvedOutput),
+
+      libraryTarget: 'commonjs2',
+    },
+    plugins: [
+      new I18nPlugin(...pluginOpts),
+    ],
+  });
+
+  return new Promise((resolve, reject) => {
+    compiler.run((err, stats) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+      // stats.compilation.errors contains errors and warnings produced by plugin itself
+      if (stats.compilation.errors.length) {
+        reject(stats.compilation.errors[0]);
+        return;
+      }
+
+      resolve(read(resolvedOutput).then((raw) => {
+        return ({
+          file: resolvedOutput,
+          raw,
+          stats,
+        });
+      }));
+    });
+  });
+}
+
+function read(filepath) {
+  return new Promise((resolve, reject) => {
+    readFile(filepath, 'utf8', (err, data) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      resolve(data);
+    });
+  });
+}
+
+export function requireUncache(modulePath) {
+  delete require.cache[require.resolve(modulePath)];
+  return require(modulePath); // eslint-disable-line import/no-dynamic-require, global-require
+}
```

Comment:
Something else to keep in mind, as an organization we are trying to retain all of Jest's functionality while still testing through Webpack.

While I understand this is a WIP, it would be worth taking a look at https://github.com/webpack-contrib/css-loader/tree/new-loader/test - this is the general direction we are taking with all of our test implementations.

It allows execution of tests through webpack to catch any edge cases missed by unit testing while maintaining Jest's snapshots, coverage, incremental compile / tests & async features.

If you want or need a hand just @ me